probe-rs/src/session.rs

```diff
@@ -432,6 +423,111 @@ impl Session {
         Ok(interface)
     }
 
+    fn reattach_arm_interface(
```

Comment:
Why does this swap the real interface for a dummy one and then gets the dummy probe from the dummy, detaches and reattaches on the fake probe but not actually on the real probe? Was this just an experiment?

Reply:
It creates a FakeProbe first, then swaps that for the real interface so `&mut interface` becomes the fake one, and `tmp_interface` becomes the real one.  Then we have owned access to the real one instead of &mut and we can perform the re-attach. finally we swap back the real and fake at the end of the function so the caller never sees the difference.

This is a bit hacky and I'd be happy to solve this another way. But we need to have access to `interface` by owned value in order to re-attach it and we basically never have that after the session is created.

Reply:
Ah lol I was too dumb to read correctly, thanks! Yeah, I see the issue. Hmm. I mean it's logically sounds, so I am fine with it. But an explanation why it is done would be nice (the part about needing owned access).

Reply:
NP,  will add